src/modules/popupModal.js

```diff
@@ -0,0 +1,80 @@
+import getDetail from './getDetail.js';
+import fetchComments from './fetchComments.js';
+import submitComment from './submitComment.js';
+import commentCounter from './commentCounter.js';
+
+const newModal = document.getElementById('new-modal');
+const newClosebtn = document.getElementById('new-close');
+const newItemSection = document.getElementById('new-itemSection');
+const newCommentSection = document.getElementById('new-commentSection');
+const newCommentsHeading = document.getElementById('new-commentsHeading');
+const newBtnAddComment = document.getElementById('new-btnAddComment');
+const newUsernameInput = document.getElementById('new-username');
+const newCommentInput = document.getElementById('new-comment');
+
+const clearElements = (elements) => {
+  elements.forEach((element) => {
+    element.innerHTML = '';
+  });
+};
+
+const closeModal = () => {
+  newModal.style.display = 'none';
+  clearElements([newItemSection, newCommentsHeading, newCommentSection]);
+};
+
+const displayItemDetail = async (id, data, comments) => {
+  newModal.style.display = 'flex';
+
+  newItemSection.innerHTML = `
+    <img class='img-beautiful' src='${data.strMealThumb}'>
+    <h2>${data.strMeal}</h2>
+    <p class='p'>${data.strInstructions}</p>
+  `;
+
+  if (comments.length > 0) {
+    const count = commentCounter(comments);
+    newCommentsHeading.innerHTML = `<p class='commentCountTitle'>Comments(${count})</p>`;
+    newCommentSection.innerHTML = comments
+      .map(
+        (comment) => `
+          <div class="commentBox-beautiful">
+            <p class="comment-name">${comment.username}:</p>
+            <p class="commentText-beautiful">${comment.comment}</p>
+            <p class="date-beautiful">${comment.creation_date}</p>
+          </div>
+        `,
+      )
+      .join('');
+  }
+
+  newBtnAddComment.addEventListener('click', async () => {
+    const username = newUsernameInput.value;
+    const comment = newCommentInput.value;
+
+    if (username !== '' && comment !== '') {
+      await submitComment(id, username, comment);
+      newUsernameInput.value = '';
+      newCommentInput.value = '';
+      closeModal();
+    }
```

Comment:
- [ ] I tested your app by commenting on different cards. When I type text in the input fields and press the add comment button, the form does nothing, but when I refresh the page I can see my comment. This issue is not a good user experience. Kindly troubleshoot and fix this issue, You can delete the app on your machines and clone it from GitHub.

![comment](https://github.com/MohammadYaser/Javascript-capstone/assets/53124548/40c24cf5-6072-4ee7-90b2-db40ade6eac0)